RZAssemblage/RZAssemblage.h

```diff
@@ -6,14 +6,131 @@
 //  Copyright (c) 2015 Raizlabs. All rights reserved.
 //
 
-#import <UIKit/UIKit.h>
+#import <Foundation/Foundation.h>
 
-//! Project version number for RZAssemblage.
 FOUNDATION_EXPORT double RZAssemblageVersionNumber;
-
-//! Project version string for RZAssemblage.
 FOUNDATION_EXPORT const unsigned char RZAssemblageVersionString[];
 
-// In this header, you should import all the public headers of your framework using statements like #import <RZAssemblage/PublicHeader.h>
+#import "RZAssemblageChangeSet.h"
+
+OBJC_EXTERN NSString *const RZAssemblageUpdateKey;
+
+@class NSFetchedResultsController;
+
+typedef NS_OPTIONS(NSUInteger, RZAssemblageEnumerationOptions) {
+    RZAssemblageEnumerationNoOptions = 0,
+    RZAssemblageEnumerationBreadthFirst = 1 << 1, // Default is depth first
+    RZAssemblageEnumerationIncludeNilRepresentedObject = 1 << 2, // Default is to skip nil representations
+};
+
+@protocol RZAssemblageDelegate;
+
+/**
+ *  RZAssemblage allows objects to be composed into trees of objects that can be
+ *  accessed via NSIndexPath.  It also allows the mutation of the tree to percolate
+ *  up the tree, and sync with supported UI elements.
+ */
+@interface RZAssemblage : NSObject
+
++ (RZAssemblage *)assemblageForArray:(NSArray *)array;
++ (RZAssemblage *)assemblageForArray:(NSArray *)array representedObject:(id)representedObject;
++ (RZAssemblage *)joinedAssemblages:(NSArray *)array;
++ (RZAssemblage *)assemblageTreeWithObject:(id)object arrayKeypaths:(NSArray *)keypaths;
++ (RZAssemblage *)assemblageTreeWithObject:(id)object arrayTreeKeypath:(NSArray *)keypaths;
++ (RZAssemblage *)assemblageWithObject:(id)object leafKeypaths:(NSArray *)keypaths;
++ (RZAssemblage *)assemblageForFetchedResultsController:(NSFetchedResultsController *)frc;
```

Comment:
These need docs, like we talked about. If any of these methods seem unimportant for now, remove them from public header and save for potential later release.